arbor/backends/gpu/reduce_by_key.hpp

```diff
@@ -1,12 +1,58 @@
 #pragma once
 
 #include <cstdint>
-#include "cuda_atomic.hpp"
-#include "cuda_common.hpp"
+#include "gpu_atomic.hpp"
+#include "gpu_common.hpp"
 
 namespace arb {
 namespace gpu {
 
+// double shuffle for AMD devices
+#ifndef __NVCC__
+__device__ __inline__ double shfl(double x, int lane)
```

Comment:
This needs to be specialised with a clang-cuda version that uses the `_sync` versions with `mask`.